### PR TITLE
Eliminate SortedSetDisk when building row diffs

### PR DESCRIPTION
**Pull request by danieldanciu**
- simply use a file merger, most of the functionality of SSD (ChunkedWaitQueue, de-duping) is not used anyway
 - I added the option to disable intermediate merging in SSD, even though in the end we are not using it; I think it's useful to have it.

### CODE REVIEW

File: metagraph/src/common/file_merger.hpp [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        return result;
+    }
+
+    inline bool empty() const { return els.empty(); }
+
+  private:
+    // elements stored as a min-heap
+    std::vector<value_type> els;
+    Compare compare_ = Compare();
+};
```
Comment (karasikov) [inline metagraph/src/common/file_merger.hpp:60]: Why do we need yet another merger? Why not use the ones from Elias Fano mergers?

File: metagraph/src/common/sorted_sets/sorted_set_disk_base.hpp [hunk 2]
```diff
@@ -182,18 +185,15 @@ class SortedSetDiskBase {
     ChunkedWaitQueue<T> merge_queue_;
 
     /**
-     * Thread pool for doing the "level 1" merging, i.e. merging #MERGE_L1_COUNT chunk
+     * Thread pool for doing the "level 1" merging, i.e. merging #merge_count_ chunk
      * files at a time, while new files are still being added to the data structure.
      */
-    ThreadPool async_merge_l1_ = ThreadPool(1, 100);
+    ThreadPool async_merge_l1_;;
```
Comment (karasikov) [inline metagraph/src/common/sorted_sets/sorted_set_disk_base.hpp:191]:
```suggestion
    ThreadPool async_merge_l1_;
```

**Review by karasikov — APPROVED** [review]

File: metagraph/src/common/file_merger.hpp (hunk 1 above)
Comment (karasikov) [inline metagraph/src/common/file_merger.hpp:60]: Did you benchmark this thing? Is this faster than the same thing from `elias_fano_file_merger.hpp`?

Comment (danieldanciu) [reply metagraph/src/common/file_merger.hpp:60]: I wanted a simpler merger that doesn't compress/decompress data, as in this case disk space is not an issue. But we can totally use the EF encoders, and the corresponding merger, too.

Comment (danieldanciu) [reply metagraph/src/common/file_merger.hpp:60]: I benchmarked it back in the time when I introduced the Elias-Fano encoder. I remember seeing no significant performance penalty with EF, using the Google Cloud Network disks. For SSDs I don't think EF can beat dumping the data directly, so I preferred this version. I can add a TODO: to do a proper benchmarking once the paper is out and time is not an issue.

Comment (karasikov) [reply metagraph/src/common/file_merger.hpp:60]: There is a `MergeHeap` there with exactly the same interface. The difference is only in internals.

So I think we should benchmark both of them and select the fastest one.

**Pull request merged**